scripts/script-Archive-FB-Videos-&-Upload-to-YT/run.py

```diff
@@ -0,0 +1,134 @@
+# Import required libraries
+import os
+import google_auth_oauthlib
+from googleapiclient import discovery
+import googleapiclient.errors
+import googleapiclient.http
+import json
+
+# Define the required permissions for YouTube API
+SCOPES = ["https://www.googleapis.com/auth/youtube.upload"]
+TOKEN_FILE = 'token.json'
+CLIENT_SECRETS_FILE = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'client.json')
+DEFAULT_TITLE = "Live video-Sociedad de Astronomia del Caribe"
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+SAMPLE_DIR = os.path.join(SCRIPT_DIR, 'facebook-sociedadastronomia')
+JSON_FILE = os.path.join(
+    SAMPLE_DIR,
+    "this_profile's_activity_across_facebook",
+    'live_videos',
+    'live_videos.json'
+)
+
+# Returns an authenticated YouTube API client
+def authenticate_youtube():
+    # Allow OAuth2 over HTTP (not recommended for production)
+    os.environ["OAUTHLIB_INSECURE_TRANSPORT"] = "1"
+
+    # Remove existing token file if it exists to force new authentication
+    if os.path.exists(TOKEN_FILE):
+        os.remove(TOKEN_FILE)
+
+    # Path to your OAuth 2.0 client credentials file
+    # This file should be downloaded from Google Cloud Console
+    client_secrets_file = CLIENT_SECRETS_FILE
+
+    # Create OAuth 2.0 flow and run local server for authentication
+    flow = google_auth_oauthlib.flow.InstalledAppFlow.from_client_secrets_file(
+        client_secrets_file, SCOPES)
+    credentials = flow.run_local_server()
+
+    # Build and return the YouTube API client
+    youtube = discovery.build(
+        "youtube", "v3", credentials=credentials)
+
+    return youtube
+
+def upload_video(youtube):
```

Comment:
there's too much logic in this method
try to refactor to reduce the complexity of this code and improve separation of concerns